src/main/java/blockchains/iaas/uni/stuttgart/de/restapi/Controllers/PluginManagerController.java

```diff
@@ -0,0 +1,128 @@
+package blockchains.iaas.uni.stuttgart.de.restapi.Controllers;
+
+import blockchains.iaas.uni.stuttgart.de.config.ObjectMapperProvider;
+import blockchains.iaas.uni.stuttgart.de.management.BlockchainPluginManager;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.glassfish.jersey.media.multipart.FormDataContentDisposition;
+import org.glassfish.jersey.media.multipart.FormDataParam;
+import org.pf4j.PluginWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import javax.ws.rs.core.UriInfo;
+
+@Path("plugin-manager")
+public class PluginManagerController {
+    private static final Logger log = LoggerFactory.getLogger(PluginManagerController.class);
+
+    @Context
+    protected UriInfo uriInfo;
+
+    @POST
+    @Path("/upload-plugin")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response uploadJar(@FormDataParam("file") InputStream uploadedInputStream,
+                              @FormDataParam("file") FormDataContentDisposition fileDetails) {
+
+        BlockchainPluginManager blockchainPluginManager = BlockchainPluginManager.getInstance();
+        String fileName = fileDetails.getFileName();
+        log.info("Received file {}", fileName);
+        String uploadedFileLocation = blockchainPluginManager.getPluginPath() + "/" + fileDetails.getFileName();
+        java.nio.file.Path filePath = Paths.get(uploadedFileLocation);
+        if (Files.exists(filePath)) {
+            log.error("Received file {} already exists in plugins directory.", fileName);
+            return Response.status(Response.Status.BAD_REQUEST).entity("File already exists with same name.").build();
+        }
+        writeToFile(uploadedInputStream, uploadedFileLocation);
+
+        blockchainPluginManager.loadJar(filePath);
+
+        return Response.ok().build();
+    }
+
+    @POST
+    @Path("/enable-plugin")
+    public Response enablePlugin() {
+        String pluginId = this.uriInfo.getQueryParameters().getFirst("plugin-id");
+        BlockchainPluginManager.getInstance().enablePlugin(pluginId);
+        return Response.ok().build();
+    }
+
+    @POST
+    @Path("/start-plugin")
+    public Response startPlugin() {
```

Comment:
We can start the plugin immediately after uploading as well. But, right now we are not doing so because I thought `upload` api should only store the plugin jar and not start it. We can change this behavior to start immediately after upload. 

A separate api is needed to explicitly start the plugin if admin user disabled the plugin and wants to start it again.

Reply:
Let's leave it as it is for now..